src/pubsub.js

```diff
@@ -203,6 +209,46 @@ class BasicPubSub extends Pubsub {
     this._handleRpcControl(peer, rpc)
   }
 
+  /**
+   * Validates the given message.
+   * @param {Peer} peer
+   * @param {rpc.RPC.Message} message
+   * @returns {Promise<Boolean>}
+   */
+  async validate (peer, message) {
+    const isValid = await super.validate(message)
+    if (!isValid) {
+      return false
+    }
+    for (const topic of message.topicIDs) {
+      const validatorFn = this.topicValidators.get(topic)
+      if (validatorFn) {
+        const result = validatorFn(topic, peer, message)
+        if (!this._processTopicValidatorResult(topic, peer, message, result)) {
```

Comment:
return the function result instead of the condition?

Reply:
process the results all at once w a reduce instead of bailing after the first falsy processed result?

Reply:
I think that this was changed according to what I was thinking :)